dist/tools/urlcheck/check.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/sh
+
+# Copyright 2020 Gunar Schorcht <gunar@schorcht.net>
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+if [ -z "$(which wget)" ]; then
+    echo "wget is not installed"
+    exit
+fi
+
+: "${RIOTBASE:=$(cd $(dirname "$0")/../../../; pwd)}"
+cd "$RIOTBASE"
+
+: "${RIOTTOOLS:=${RIOTBASE}/dist/tools}"
+. "${RIOTTOOLS}"/ci/changed_files.sh
+
+FILES=""
+while [ $# -gt 0 ]; do
+    FILES="${FILES} $1"
+    shift
+done
+
+FILEREGEX='\.([Hh]|[Hh]pp|txt|md)$'
+EXCLUDE='^(.+/vendor/|.+/bin/|.+dist/tools/|dist/tools/coccinelle/include|boards/common/msba2/tools/src)'
+
+if [ -z "${FILES}" ]; then
+    FILES=$(FILEREGEX=${FILEREGEX} EXCLUDE=${EXCLUDE} changed_files)
+else
+    FILES=$(echo "${FILES}" | grep -E "${FILEREGEX}" | grep -vE "${EXCLUDE}")
+fi
+
+if [ -z "${FILES}" ]; then
+    exit
+fi
+
+URL_REGEX='(?=https://|http://)[^ )\]>\"]+'
```

Comment:
Can we add a way to provide excluded patterns ? There's one broken link reported which is on purpose, since it's an example url in `sys/include/net/sock/util.h: `https://host.name:1234/url/path`